### DISCUSSION TITLE
Smooth transformation of 3D CAD models in UMEP

### DISCUSSION THREAD

**Discussion by oraleks** [UMEP-dev/UMEP#358] [Ideas]
A feature that will enable smooth import of 3D digital models produced in CAD software used for urban design (e.g. Rhino, Sketchup, AutoCAD) will greatly enhance the operability of UMEP as an urban design tool, and especially for choosing between different future design options.

**Comment by rarygit** [comment]
QGIS already has the capability to smoothly import 3D models in the GeoJSON format.
I would suggest an alternative issue:  "Smooth export of 3D CAD format (GeoJSON) from Rhino and Autocad".

As an example, there is a description at this blog: "How to Create a 3d buildings view in QGIS"
The blog has been written by a commercial data provider. Nevertheless it is informative.
https://3dbuildings.medium.com/how-to-create-a-3d-buildings-view-in-qgis-8351ea5a3138
3dbuildings Data Format:
https://3dbuildings.com/documentation/format/
3dbuilding.com examples:
https://3dbuildings.medium.com/global-height-84a8c2f18e7a

Rhino: Working with and Exporting GeoJSON data
https://toolbox.decodingspaces.net/working-with-and-exporting-geospatial-data/

AutoCAD: Import/export GeoJSON
https://forums.autodesk.com/t5/autocad-map-3d-forum/coordinate-convertion-wgs84-and-convert-to-geojson/td-p/9359333
https://apps.autodesk.com/CIV3D/en/Detail/Index?appLang=en&id=8738886170670550904&os=Win32_64&us_oa=forums-us&us_si=770cfdd6-aad9-4fd7-a299-224086d84749&us_st=geojson

**Comment by biglimp** [comment]
The challenge here is not to convert DWG/DXF to other vector-based format but to convert the 3D vector information into raster datasets that could be used in UMEP. In particular, I point towards >LOD2 vector polygon objects such as e.g. tilting roof structures. You need to calculate a height for each raster pixel on a sloping plane where only four points (xyz) is known. I still have only seen one tool that can deal with this (PointCloudCombiner in Safe FME). However, this is not undoable but so far not something that I have seen in an open source tool. If someone create (or finds) such a tool, please let me know.

**Reply by gusbacos** [reply]
@biglimp  _"In particular, I point towards >LOD2 vector polygon objects such as e.g. tilting roof structures. You need to calculate a height for each raster pixel on a sloping plane where only four points (xyz) is known. I still have only seen one tool that can deal with this (PointCloudCombiner in Safe FME). However, this is not undoable but so far not something that I have seen in an open source tool. If someone create (or finds) such a tool, please let me know."_

I think WhiteboxTools LidarDigitalSurfaceModel (https://www.whiteboxgeo.com/manual/wbt_book/available_tools/lidar_tools.html#lidardigitalsurfacemodel) could preform this task. It is possible to run standalone directly in python, or install in the processing toolbox in QGIS. 

Down below is a screenshot from DSM created from a Raw-Pointcloud using only default settings in the QGIS-GUI for WhiteboxTools

![image](https://user-images.githubusercontent.com/44983335/151790027-01d4cffc-d638-42a8-92c1-80c822c91ea8.png)

**Reply by biglimp** [reply]
Thanks Oskar, but I think this is not what I am referring to. This tool converts Lidar to DSMs. I cannot find anything about the conversion between 3D vector (LOD2) to raster, but maybe I am wrong. I attach an image of vector LOD2 roof structures that I am thinking of. LOD1 is no problem since that height can be stored in attribute data (as in the e.g. DSMGenerator). One possible way would be to convert Vector to Lidar point cloud and then to DSM.

![image](https://user-images.githubusercontent.com/11908356/151794984-207eed65-e9ba-4a89-823d-a28ce20fdf73.png)
